### PR TITLE
Fix persistent halt command

### PR DESCRIPTION
**Pull request by preston-rogers**
This pull request is a follow up of the previous pull request for fix-persistent-halt-command.

This pull request solves the problem using a flag that ensures that the ELMO has time to switch to operation enabled before processing halt commands.

Thus far, this implementation is untested.

### CODE REVIEW

**Comment by preston-rogers** [comment]
@d-loret Regarding this implementation, I am only concerned with what the value of "enabling_operation" would be at initialization of the state struct.
By default, I believe booleans are undefined by value. Would this be an issue?

File: src/jsd_egd_types.h [hunk 1]
```diff
@@ -294,6 +294,7 @@ typedef struct {
   uint32_t motor_rated_current;  ///< CL[1] in mA
 
   // user input
+  bool                        enabling_operation;
```
Comment (d-loret) [inline src/jsd_egd_types.h:297]: Let's initialize the flag at the end of `jsd_egd_init` where we initialize other variables of the state.

**Review by d-loret — CHANGES_REQUESTED** [review]
This follows what I suggested but we need to turn off the flag when transitioning into the fault state. See also my comment on where we need to initialize the flag.
 I commented only in the EGD files, but the comments apply to the EPD files too.

File: src/jsd_egd_types.h (hunk 1 above)
Comment (preston-rogers) [reply src/jsd_egd_types.h:297]: Done

File: src/jsd_egd.c [hunk 2]
```diff
@@ -1255,8 +1258,8 @@ void jsd_egd_process_state_machine(jsd_t* self, uint16_t slave_id) {
       }
       break;
     case JSD_ELMO_STATE_MACHINE_STATE_OPERATION_ENABLED:
-
-      state->pub.fault_code = JSD_EGD_FAULT_OKAY;
+      state->enabling_operation  = false;
```
[the remark this answers is not in the document]
Comment (preston-rogers) [reply src/jsd_egd.c:1264]: So I'm guessing you determined that setting the new_reset to false when the platinum drive enters a fault state is correct (as per the TODO comment)?
I have just made these changes.

File: src/jsd_egd.c [hunk 3]
```diff
@@ -1334,6 +1336,7 @@ void jsd_egd_process_state_machine(jsd_t* self, uint16_t slave_id) {
   }
 
   state->new_motion_command = false;
+  if (!state->enabling_operation) state->new_halt_command = false;
```
[the remark this answers is not in the document]
Comment (preston-rogers) [reply src/jsd_egd.c]: Done

**Review by d-loret — APPROVED** [review]
Looks good to me.
Please test on EGD before merging.

**Pull request merged**